fw-child/resources/js/child-functions.js

```diff
@@ -233,19 +233,37 @@ const $ = jQuery;
       });
     }
 
+    let pinned_item = null;
     let scroll_offset = 0;
 
     if ($('#floating-header').length) {
       scroll_offset += $('#floating-header').outerHeight();
     }
 
     if ($('.query-page #control-bar-tabs').length) {
-      new $.Zebra_Pin($('#control-bar-tabs'), {
+      pinned_item = new $.Zebra_Pin($('#control-bar-tabs'), {
         top_spacing: scroll_offset,
-        contained: true,
+        contain: true,
       });
     }
 
+    let resize_observer = new ResizeObserver( function() {
```

Comment:
I think this whole JS can be optimized.
I think you don't need the ResizeObserver, because you are already using $(window).on('resize'... and the #control-bar will not get resized if the window is not resized. Am I right?

You are checking if pinned_item != null in the ResizeObserver function but not in the window resize.
Wrap your repetitive code into small functions to avoid the repetition.

Please use camel case instead of snake case for code consistency.

Reply:
@domcode-ew the code was refactored to use only the ResizeObserver based on the conversation we had. Can you check again? Thanks!